annotation of pod/ns/dep with an invalid EPG will be shown as a fault

Bhavanaashok33 opened the pull request:
Fault will be raised against ns/pod/deployment if the epg doesnt exist on APIC
Added unit tests 
Tested the changes on fabric 
Signed-off-by: bashokba <bashokba@cisco.com>

tomflynn commented:
Can you enhance the apic_types_test.go to cover the new types you've added.
Can you also update the UTs so it tests the fault creation? coveralls shows those lines aren't being covered by UTs currently

Bhavanaashok33 commented:
I have addressed all the review comments. wrt fault creation test on APIC, I have simulated a mock APIC(local http server) that accepts the REST API and partially mock's the APIC.  Currently Testing is it process. please don't review the code yet..

Bhavanaashok33 commented:
Please go ahead and review the code! 
Enhanced the UTs covering fault creating on mock APIC.

jayaramsatya commented on pkg/controller/annotation_test.go:
```diff
@@ -0,0 +1,302 @@
+	cert bool
+}
+
+type socketHandler struct {
```
you can reference this structure apicapi, please check where ever you can.

jayaramsatya commented on pkg/controller/pods.go:
```diff
@@ -92,29 +101,39 @@ func (cont *AciController) checkIfEpgExistPod(pod *v1.Pod) {
-		faultCode := strconv.Itoa(10)
-		severity := "major"
-		cont.createFaultInst(key, desc, faultCode, severity)
+	if len(pod.ObjectMeta.Annotations[metadata.EgAnnotation]) > 0 {
```
Please change this line 
epGroup, ok  := pod.ObjectMeta.Annotations[metadata.EgAnnotation]
if ok {
  cont.handleEpgAnnotationUpdate(key, 10, major, pod.Name, epGroup)
}

jayaramsatya commented on pkg/controller/subnetcache.go:
```diff
@@ -132,7 +130,7 @@ func (cont *AciController) BuildSubnetDnCache(dn string, aciVrfDn string) {
 	sort.Strings(vrfEpgDns)
 	cont.log.Debug("aciVrfEpgDns: ", vrfEpgDns)
-	cachedVRFDns = vrfEpgDns
+	cont.cachedVRFDns = vrfEpgDns
```
can you check if you need to lock this?

jayaramsatya commented on pkg/controller/pods.go:
```diff
@@ -31,6 +32,15 @@ import (
 )
 
+const (
+	cleared  = iota
```
please give the name to the severity

jayaramsatya commented on pkg/controller/namespaces.go:
```diff
@@ -126,24 +125,16 @@ func (cont *AciController) namespaceDeleted(obj interface{}) {
-		faultCode := strconv.Itoa(11)
-		severity := "critical"
-		cont.createFaultInst(nskey, desc, faultCode, severity)
+	if len(namespaceobj.ObjectMeta.Annotations[metadata.EgAnnotation]) > 0 {
```
change it same as above  comment

jayaramsatya reviewed (approved):
LGTM
